include/lug/System/Exception.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <exception>
+#include <cstdint>
+#include <string>
+#include <lug/System/Export.hpp>
+
+
+#if defined(LUG_SYSTEM_WINDOWS)
```

Comment:
Why not doing this near the other macro? And you are polluting the context, use undef to remove this useless macro after using it!